index.js

```diff
@@ -68,7 +68,37 @@ function toggleColor(element) {
   }
 }
 
-/***** Deliverable 2 *****/
+const header = document.querySelector("#header")
+
+header.addEventListener("click", function(){
+  toggleColor(header)
+})
 
+/***** Deliverable 2 *****/
+const button = document.querySelector('.like-button')
+const pLike = document.querySelector('.likes')
 
+button.addEventListener('click', function(){
+  const likes = parseInt(pLike.textContent)
+  pLike.textContent = likes + 1 + " Likes"
+})
 /***** Deliverable 3 *****/
+const form = document.querySelector("#new-goal-form")
+const goalLinkForm = form.querySelector("#goal-link")
+const goalDescription = form.querySelector("goal-description")
+
+form.addEventListener("submit", function(event){
+  event.preventDefault()
+  const link = event.target.link.value
+  const description = event.target.description.value
+
+  const goalId = document.querySelector("#goals").lastChild.dataset.id
+  const newId = parseInt(goalId) + 1
+  const newGoal = {
+    id: newId,
+    link: link,
+    description: description
```

Comment:
Nie work! What if the ul#goals doesn't have any children li elements? Then `goaldId` would be undefined and the ID incrementation wouldn't work properly. You could add a conditional to check for that edge case. You could also pull the latest ID from the `goals` array in the `player` object in `data.js` instead to create the new id. If you want to shorten your code or save on variable declarations, you could also create your object like so:

```Javascript
  const newGoal = {
    id: parseInt(goalId) + 1
    link: event.target.link.value,
    description: event.target.description.value
  }
```